drivers/stepper/step_dir_stepper_controller.c

```diff
@@ -0,0 +1,264 @@
+/*
```

Comment:
Are there common functions here which could be used by gpio stepper controller as well?

I actuallyhad something like this in mind. Do you think renaming this to gpio_stepper_common is more apt? Since step dir steppers can also be driven using gpio toggle?

Reply:
The overall structure is very similar yes, but I am not sure how to best fit in the coil state table without being messy.

Reply:
Yes, exactly. I guess the data structures can be reused and some of the setter/getter methods as well.